app/league/[leagueId]/entry/[entryId]/week/Week.test.tsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import Alert from '@/components/AlertNotification/AlertNotification';
+import { AlertVariants } from '@/components/AlertNotification/Alerts.enum';
+import { toast } from 'react-hot-toast';
+import { onWeeklyPickChange } from './WeekHelper';
+import { createWeeklyPicks } from '../../../../../../api/apiFunctions';
+import { parseUserPick } from '../../../../../../utils/utils';
+
+jest.mock('../../../../../../api/apiFunctions', () => ({
+  createWeeklyPicks: jest.fn(),
+}));
+
+jest.mock('react-hot-toast', () => ({
+  toast: {
+    custom: jest.fn(),
+  },
+}));
+
+describe('Week', () => {
+  const data = {
+    target: { value: 'Browns' },
+    preventDefault: jest.fn(),
+    stopPropagation: jest.fn(),
+  } as any;
+  const NFLTeams = [{ teamName: 'Browns', teamId: 'mockId' }];
+  const user = { id: 'mockUserId', email: 'email@example.com', leagues: [] };
+  const entry = 'mockEntry';
+  const league = 'mockLeague';
+  const week = 'mockWeek';
+  const updateWeeklyPicks = jest.fn();
+  const setUserPick = jest.fn();
+  interface IUserResults {
+    [key: string]: any; // Adjust the 'any' to match the structure of your user results
```

Comment:
Same here. This is opening up a can of worms. Because we strictly type everything, there should be a type that matches this.